src/CoreLib/network.fs

```diff
@@ -286,6 +286,10 @@ type [<AllowNullLiteral>] NetworkCommandQueue internal () as x =
 
     let eVerified = new ManualResetEventSlim(false)
 
+    let eAsyncSend = new ManualResetEventSlim(true)
```

Comment:
Do we expect to wait on eAsyncSend using WaitForSingleObject (or WaitForHandle)? 
The current code may have issues if there are multiple AsyncSends outstanding. Say we have AsyncSend1, AsyncSend2, and x.StartConnection(), AsyncSend1 proceeds, while AsyncSend2 and StartConnection gets blocked on eAsyncSend.Wait(), when AsyncSend1 finishes, both StartConnection  and AsyncSend2 can proceed at the same time. Will this causes an issue?

It looks to me that it might be simpler to convert eAsyncSend to an object, and just lock the object using lock(eAsyncSend). We will avoid an object to dispose. lock(o) uses Monitor.Enter, which uses SpinLock and keyed event internally, and may lead to better efficiency.